astropy/wcs/wcs.py

```diff
@@ -389,6 +389,9 @@ def __init__(self, header=None, fobj=None, key=' ', minerr=0.0,
                     header_string = header
             elif isinstance(header, fits.Header):
                 header_string = header.tostring().rstrip()
+            elif type(header) == fits.ImageHDU or type(header) == fits.PrimaryHDU:
```

Comment:
This should be written `isinstance(header, (fits.ImageHDU, fits.PrimaryHDU))`, or you can kill two birds with one stone by going with their common base class (which may be broader anyways): `isinstance(header, fits.hdu.image._ImageBaseHDU)`.